include/fiction/algorithms/physical_design/apply_gate_library.hpp

```diff
@@ -143,25 +152,31 @@ class apply_gate_library_impl
                         relative_to_absolute_cell_position<GateLibrary::gate_x_size(), GateLibrary::gate_y_size(),
                                                            GateLyt, CellLyt>(gate_lyt, t, cell<CellLyt>{0, 0});
 
-                    const auto gate = GateLibrary::template set_up_gate<GateLyt, CellLyt, Params>(gate_lyt, t, params);
-
-                    assign_gate(c, gate, n);
+                    assign_gate(
+                        c, GateLibrary::template set_up_gate<GateLyt, CellLyt, Params>(gate_lyt, t, params, lyt), n);
                 }
 #if (PROGRESS_BARS)
                 // update progress
                 bar(i);
 #endif
             });
 
-        // perform post-layout optimization if necessary
-        if constexpr (has_post_layout_optimization_v<GateLibrary, CellLyt>)
-        {
-            GateLibrary::post_layout_optimization(cell_lyt);
-        }
-
         // if available, recover layout name
         cell_lyt.set_layout_name(get_name(gate_lyt));
 
+        if constexpr (is_sidb_defect_surface_v<CellLyt>)
+        {
+            if (lyt.has_value())
+            {
+                // due to issue with windows-2019 Visual Studio 16 2019 and v142.
+                auto copy_lyt = cell_lyt.clone();
+                // add defects to the circuit.
+                lyt.value().foreach_sidb_defect([this, &copy_lyt](const auto& def)
+                                                { copy_lyt.assign_sidb_defect(def.first, def.second); });
```

Comment:
Why adding defects?

Reply:
beacuse they are not added by ``assign_gate(c, GateLibrary::template set_up_gate<GateLyt, CellLyt, Params>(gate_lyt, t, params, lyt), n);``. Hence, they need be added.

Reply:
I see. Maybe I was just confused by the comment "adding defects" that seemed like they were actual additions out of nowhere. Instead, they were just copied over from the other layout. Maybe we can make that clearer in the comment?